src/libs/pr.ts

```diff
@@ -1,9 +1,41 @@
 import * as core from '@actions/core';
 import { context, getOctokit } from '@actions/github';
-import * as dedent from 'dedent';
+import AnsiToHtml from 'ansi-to-html';
+import dedent from 'dedent';
 import invariant from 'ts-invariant';
 import { Config } from '../config';
 
+function ansiToHtml(
+  message: string,
+  maxLength: number,
+): [string, boolean] {
+  /**
+   *  Converts an ansi string to html by for example removing color escape characters.
+   *  message: ansi string to convert
+   *  maxLength: Maximum number of characters of final message incl. HTML tags
+   *
+   *  return message as html and information if message was trimmed because of length
+   */
+  const convert = new AnsiToHtml();
+  let trimmed = false;
+
+  let htmlBody: string = convert.toHtml(message);
+
+  // Check if htmlBody exceeds max characters
+  if (htmlBody.length > maxLength) {
+
+    // trim input message by number of exceeded characters
+    const dif: number = htmlBody.length - maxLength;
+    message = message.substring(0, message.length - dif);
```

Comment:
Hrm trimming it this way doesn't seem quite correct when we have HTML here, as we're risking trimming it in the middle of an HTML tag.  Not quite sure of the top of my head what the best way to go about that is.

Maybe we could convert it to HTML, check how much longer than allowed the message is, trim the original message by that much and then re-convert it to HTML?  Though that might shorten it by too much.  Maybe sort of a binary search style would work best to see how much we can fit into the max length?

Reply:
Actually I thought I'm doing exactly what you proposed:
1. Converting the `message` to html `htmlBody`
2. In case this result html is to long I check on how much `dif`
3. The original plain `message` is than trimmed by the amount of characters and converted to html again.
This might result in a bit shorter html message, which was adaptable for me keeping the complexity in mind to find the exact maximum html length.

Could you please cross check again and let me know if I missed something in the code that leads to a behavior other than described here.

Reply:
Oh doh, sorry.  I totally misread this :facepalm: 

What you have here looks good.